autogpts/autogpt/autogpt/agents/components.py

```diff
@@ -0,0 +1,40 @@
+from typing import Callable, Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class Single(Generic[T]):
+    """A wrapper for a single result (non-pipeline) of a component function."""
+
+    def __init__(self, value: T):
+        self.value = value
+
+
+class Component:
+    run_after: list[type["Component"]] = []
+    enabled: Callable[[], bool] | bool = True
+    disabled_reason: str = ""
```

Comment:
The `Component` class, can specify types it should be run after, and a callable for checking if it's enabled, with optional reason.